src/ddspdrum/torchmodule.py

```diff
@@ -486,12 +529,164 @@ def _forward(self, control_in: T, audio_in: T) -> T:
         assert (control_in >= 0).all() and (control_in <= 1).all()
 
         if (audio_in <= -1).any() or (audio_in >= 1).any():
-            normalize(audio_in)
+            util.normalize(audio_in)
 
-        audio_in = fix_length(audio_in, len(control_in))
+        audio_in = util.fix_length(audio_in, len(control_in))
         return control_in * audio_in
 
 
+class TorchNoise(TorchSynthModule):
+    """
+    Adds noise to a signal
+
+    Parameters
+    ----------
+    ratio (float): mix ratio between the incoming signal and the produced noise
+    **kwargs: see TorchSynthModule
+    """
+
+    def __init__(
+            self,
+            ratio: float = 0.25,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.add_parameters(
+            [
+                TorchParameter(
+                    value=ratio,
+                    parameter_name="ratio",
+                    parameter_range=ParameterRange()
```

Comment:
`(0, 1)` ?

Reply:
These are default values for ParameterRange -- happy to explicitly add them though, might be helpful for readability